lib/run_loop/device_agent/client.rb

```diff
@@ -1464,6 +1464,14 @@ def launch_cbx_runner
           shutdown
         end
 
+        if device.simulator?
+          core_simulator = RunLoop::CoreSimulator.new(device, nil)
+          if core_simulator.simulator_state_requires_relaunch?
+            RunLoop.log_debug("Simulator requires relaunch; shutting down")
+            shutdown
```

Comment:
Calling `shutdown` here only terminates the running DeviceAgent server on the device.

What you want is `CoreSimulator.quit_simulator` or `CoreSimulator.terminate_core_simulator_processes`.

My best guess is that you are encountering problems with orphan CoreSimulator processes.  On our CI system, we periodically terminate core simulator processes.

https://github.com/calabash/jenkins - see the hourly and daily jobs.

CoreSimulator.terminate_simulator_processes is available on the command line:

```
run-loop simctl manage-processes
```

Reply:
`shutdown` is killing DeviceAgent server and this requires the relaunch :) I can quit_simulator as well, just didn't find the method that can be easily used from the existing context.